dashboard/src/components/AppView/AppControls.tsx

```diff
@@ -103,6 +103,39 @@ class AppControls extends React.Component<IAppControlsProps, IAppControlsState>
   private togglePurge = () => {
     this.setState({ purge: !this.state.purge });
   };
+
+  private renderUpgradeButton = () => {
```

Comment:
perhaps we should make this a stateless component since it's getting a bit more complex?